integration/e2e_test.go

```diff
@@ -88,7 +88,29 @@ func TestInPlaceUpgrade(t *testing.T) {
 	// Important: the cluster is deleted in the Cleanup function
 	// which is called after all subtests are finished.
 	t.Logf("Deleting the cluster")
+}
+
+// TestDisableDefaultCNI deploys cluster disabled  defalut CNI  .
+// The CLUSTER_DISABLE_DEFAULT_CNI_MANIFEST_FILE environment variable should point to a manifest with the target cluster
+// With post actions calico will be installed.
+func TestDisableDefaultCNI(t *testing.T) {
+	cluster_manifest_file := os.Getenv("CLUSTER_DISABLE_DEFAULT_CNI_MANIFEST_FILE")
+	if cluster_manifest_file == "" {
+		t.Fatalf("Environment variable CLUSTER_DISABLE_DEFAULT_CNI_MANIFEST_FILE is not set. " +
+			"CLUSTER_DISABLE_DEFAULT_CNI_MANIFEST_FILE is expected to hold the PATH to a cluster manifest.")
+	}
+	t.Logf("Cluster to setup is in %s", cluster_manifest_file)
 
+	setupCheck(t)
+	t.Cleanup(teardownCluster)
+
+	t.Run("DeployCluster", func(t *testing.T) { deployCluster(t, os.Getenv("CLUSTER_DISABLE_DEFAULT_CNI_MANIFEST_FILE")) })
+	t.Run("ValidateCalico", func(t *testing.T) { validateCalico(t) })
+	t.Run("DeployMicrobot", func(t *testing.T) { deployMicrobot(t) })
+	t.Run("UpgradeClusterRollout", func(t *testing.T) { upgradeCluster(t, "RollingUpgrade") })
```

Comment:
Why do you deploy microbot and perform a cluster upgrade here? This test is about verifying that the default CNI is not deployed and that a new CNI can be deployed, no?